File: programming-language-tuto-web/src/services/BlogService.js
import api from "./api";

const BlogService = {
    getAllBlogs: () => api.get("/blog"),
    getBlogById: (blogId) => api.get(`/blog/${blogId}`),
    createBlog: (formData) => api.post("/blog/createBlog", formData),
    updateBlog: (blogId, data) => api.patch(`/blogs/${blogId}`, data),
    deleteBlog: (blogId) => api.delete(`/blog/${blogId}`),

};
export default BlogService;

File: myugproject/src/main/java/org/finalugproject/myugproject/service/BlogService.java
package org.finalugproject.myugproject.service;

import org.finalugproject.myugproject.DTO.request.BlogRequest;
import org.finalugproject.myugproject.exception.BlogException;
import org.finalugproject.myugproject.model.Blog;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

public interface BlogService {

    Blog createBlog(BlogRequest request,List<MultipartFile> imageFiles);

    String deleteBlog(Long blogId) throws BlogException;

    Blog updateBlog(Long blogId,BlogRequest request,List<MultipartFile> images) throws BlogException;

    List<Blog> findByName(String blogTitle);

    Blog findById(Long blogId) throws BlogException;

    List<Blog> findAllBLogs();
}
